### DISCUSSION TITLE
Ability to Clone the App

### DISCUSSION THREAD

**Discussion by rijul007** [ReVanced/revanced-discussions#444] [Question]
I have 2-3 YT accounts for various seperate reasons. I know one can switch between accounts in the app itself but I need them all regularly and mostly I forget to switch when opening the app. I even need notifications from all 3 which isn't possible because only 1 can be active at a time.

I've tried various methods of cloning the app, but all in vain, as the clone just crashes on start itself. I've tried "App Cloner", "Lucky Patcher", other Cloning apps available on Play Store which install it in a virtual environment or something. Those even install micro G but still reVanced crashes at start. I've even tried changing package names using apk modifying tools and all but nothing worked.

I wish there could be some ability to clone the app and produce as many clones as one wants, with each instance able to run seperate accounts.

Also, Thanks for all that such talented Devs have already done for the community :)

**Comment by oSumAtrIX** [comment]
The `microg-support` patch changes the package name. You can simply modify it to create apks with different package names, install them and log in with different accounts.

**Reply by ghost** [reply]
i can add PatchOptions in my refactor so people can change the package name without having to modify the patch.

**Reply by Sculas** [reply]
@Canny1913 Sure, go ahead. You can reference this discussion as a motivation point for the PR.

**Reply by rijul007** [reply]
> The `microg-support` patch changes the package name. You can simply modify it to create apks with different package names, install them and log in with different accounts.

I'm using the latest Termux method to make it directly in the phone. Is it possible to modify the patch in there ? Or do I have to use the PC method or something. (I'm a noob :P)

**Reply by rijul007** [reply]
Okay, so wait, I got it ! I'm a dumb-dumb, but I got it after properly comprehending Dev's Answer.

I used an app called Apktool M to rename the apk, installed and voila!, Now I have 4 ReVanced apps each working with seperate accounts in one phone itself. Thanks y'all !!!!!

You need to make sure to use microg-support patch and also I changed the package name from the middle, like "app.revynced.android.youtube" & "app.revonced.android.youtube". Idk if It would work any other way or not but just mentioning the exact steps.

**Reply by rijul007** [reply]
Okay wait again, update. When I installed the clone, it worked fine in the very first launch. But now after a while, it's bugging again. What I've found is that, it does open and sign in, but only after 3 consecutive crashes at launch. So I open and crash it 3 times, but only in the 4th attempt it works. @oSumAtrIX
